components/service/experiments/src/test/java/mozilla/components/service/experiments/FlatFileExperimentStorageTest.kt

```diff
@@ -19,15 +20,54 @@ import java.nio.file.Paths
 
 @RunWith(RobolectricTestRunner::class)
 class FlatFileExperimentStorageTest {
+    private fun checkSavedExperimentsJson(experimentsJson: JSONObject) {
+        assertEquals(2, experimentsJson.length())
+        val experimentsJsonArray = experimentsJson.getJSONArray("experiments")
+        assertEquals(1, experimentsJsonArray.length())
+        val experiment = experimentsJsonArray[0] as JSONObject
+
+        assertEquals("sample-id", experiment.getString("id"))
+        assertEquals("sample-description", experiment.getString("description"))
+        assertEquals(1526991669L, experiment.getLong("last_modified"))
+
+        val buckets = experiment.getJSONObject("buckets")
+        assertEquals(0, buckets.getInt("start"))
+        assertEquals(20, buckets.getInt("count"))
+
+        val branches = experiment.getJSONArray("branches")
+        assertEquals(2, branches.length())
+        val branch1 = branches.getJSONObject(0)
+        assertEquals("branch1", branch1.getString("name"))
+        assertEquals(2, branch1.getInt("ratio"))
+        val branch2 = branches.getJSONObject(1)
+        assertEquals("branch2", branch2.getString("name"))
+        assertEquals(5, branch2.getInt("ratio"))
+
+        val match = experiment.getJSONObject("match")
+        assertEquals("es|en", match.getString("locale_language"))
+        assertEquals("sample-appId", match.getString("app_id"))
+        val regions = match.getJSONArray("regions")
+        assertEquals(listOf("US"), regions.toList<String>())
+    }
+
     @Test
     fun save() {
         val experiments = listOf(
-            Experiment("sample-id",
-                "sample-name",
-                "sample-description",
-                Experiment.Matcher("es|en", "sample-appId", listOf("US")),
-                Experiment.Bucket(20, 0),
-                1526991669)
+            testDefaultExperiment(
+                id = "sample-id",
+                description = "sample-description",
+                match = testDefaultMatcher(
+                    localeLanguage = "es|en",
+                    appId = "sample-appId",
+                    regions = listOf("US")
+                ),
+                buckets = Experiment.Buckets(0, 20),
+                branches = listOf(
+                    Experiment.Branch("branch1", 2),
+                    Experiment.Branch("branch2", 5)
+                ),
+                lastModified = 1526991669
```

Comment:
nit: magic number (do we need to know what this date is, is it significant for us to know)?